### DISCUSSION TITLE
Help with calculating zero of an expression

### DISCUSSION THREAD

**Discussion by hpvcodeDev** [mathnet/mathnet-numerics#1056] [Q&A]
Hello, I am trying to get the zeros of an expression, this is my code

```
 Func<double, double> f = (3.67500625 * x - x * x).Compile("x");
 double resultado = Bisection.FindRoot(f, 0, 4);
```
The result returned is result = 0, a value x= 0 and y = 0, but also if we evaluate manually it also has another zero at x= 3.67500625, but that second result does not return the code that I provided because, in the case of be returning more zeros in which variable would you store it. to get all zeros.

**Comment by jkalias** [comment]
The bisection method cannot return multiple roots, unless you iteratively execute it in the domain of your problem. Since the edge of your domain IS one root, it returns this one. If you tried for example `Bisection.FindRoot(f, -2, 0);` it would also return 0 as the root of your function.

![image](https://github.com/mathnet/mathnet-numerics/assets/942804/6d328ec3-dd11-4ac8-ab79-0ddba546efb8)
